app/javascript/controllers/additional_expenses_form_controller.js

```diff
@@ -0,0 +1,36 @@
+import NestedForm from 'stimulus-rails-nested-form'
+
+export default class extends NestedForm {
+  static targets = ['additionalExpense', 'target']
+
+  add (event) {
+    super.add(event)
+    this.reindexFields()
+  }
+
+  remove (event) {
+    super.remove(event)
+    this.reindexFields()
+  }
+
+  reindexFields () {
+    // Reindex fields so that nested parameters come in with consecutive
+    // indexes starting at 0
+    const containers = this.element.querySelectorAll('.expense-container')
+    containers.forEach((container, index) => {
+      container.querySelectorAll('input, select, textarea').forEach(field => {
+        const newName = field.name.replace(/\[additional_expenses_attributes\]\[\d+\]/, `[additional_expenses_attributes][${index}]`)
+        field.name = newName
+        field.id = newName.replace(/\[/g, '_').replace(/\]/g, '')
+      })
+
+      container.querySelectorAll('label').forEach(label => {
+        const forAttr = label.getAttribute('for')
+        if (forAttr) {
+          const newForAttr = forAttr.replace(/additional_expenses_attributes_\d+/, `additional_expenses_attributes_${index}`)
+          label.setAttribute('for', newForAttr)
+        }
+      })
+    })
+  }
```

Comment:
What happens if we don't reindex the fields?

Reply:
You shouldn't need a new controller at all for this. This should be doable 100% in the template and controller. Look at the documentation [here](https://www.stimulus-components.com/docs/stimulus-rails-nested-form). 

We already have the component installed, so you can skip the installation step.

Reply:
Essentially, persisted associations get their ID as their index and new records get the timestamp they were created ([line 19](https://github.com/stimulus-components/stimulus-rails-nested-form/blob/master/src/index.ts)). This ensures unique identifiers for every association in the form without needing a reindex.

Reply:
@schoork @elasticspoon thanks! I'll make these changes and push something up by the end of this week